src/datreant/core/agglimbs.py

```diff
@@ -164,48 +164,132 @@ def __str__(self):
-                value corresponding to given key
+    def __getitem__(self, keys):
+        """Get values for a given key, list of keys, or set of keys.
+
+        If *keys* is a string specifying one key (for a single category),
+        return a list of the values among all Treants (in this collection) that
+        have that category.
+
+        If *keys* is a list of keys, return a list of lists whose order
+        corresponds to the order of the elements in *keys*. Each element in
+        *keys* is a key specifying a category; each element in the output is
+        a list of the values among all Treants (in this collection) that have
+        the category specified by the respective key in *keys*.
+
+        If *keys* is a set of keys, return a dict of lists whose keys are the
+        same as those provided in *keys*; the value corresponding to each key
+        in the output is a list of values among all Treants (in this
+        collection) that have the category corresponding to that key.
+
+        Parameters
+        ----------
+        keys : str, list of str, set of str
+            Key(s) of Categories in this collection.
+
+        Returns
+        -------
+        list, list of list, dict of list
+            Values for the (single) specified category when *keys* is str.
+
+            Groupings of values, each grouping a list, where the first grouping
+            contains the values for all members of the collection corresponding
+            to the first value in *keys*, the second grouping contains values
+            for all members of the collection corresponding the second value in
+            *keys*, etc. when *keys* is a list of str.
+
+            Values in the dict corresponding to each of the provided *keys*
+            is a grouping (list) of Treants that have the Category specified by
+            that key when *keys* is a set of str.
         """
-        categories = self._dict()
-        return categories[key]
+        collection = self._collection
+        if isinstance(keys, str):
+            k = keys
+            return [m.categories[k] if k in m.categories else None
+                    for m in collection]
+        elif isinstance(keys, list):
+            return [[m.categories[k] if k in m.categories else None
+                    for m in collection]
+                    for k in keys]
+        elif isinstance(keys, set):
+            return {k: [m.categories[k] if k in m.categories else None
+                    for m in collection]
+                    for k in keys}
+        else:
+            raise TypeError("Invalid argument; argument must be" +
+                            " a string, list of strings, or set" +
+                            " of strings.")
 
     def __setitem__(self, key, value):
-        """Set value at given key.
-
-        :Arguments:
-            *key*
-                key of value to set
+        """Set the value of a category for all Treants in the collection.
```

Comment:
Good! But it should also be able to take a list of values of the same length as the number of members and set the value for each one according to that list. Basically, one could then do:

```python

b = Bundle('sprout', 'shrub', 'deku')
b.categories['bark'] = ['shaggy', 'smooth', None]
```
and this will result in `sprout` getting `{'bark': 'shaggy'}`, 'shrub' getting `{'bark': 'smooth'}`, and 'deku' getting its 'bark' category deleted, if it even existed at all.